distribution/tools/launchers/src/main/java/org/elasticsearch/tools/launchers/SystemJvmOptions.java

```diff
@@ -66,4 +70,12 @@
         );
     }
 
+    private static String maybeShowCodeDetailsInExceptionMessages() {
+        if (JavaVersion.majorVersion(JavaVersion.CURRENT) >= 14) {
```

Comment:
We should be comparing `BuildParams.compilerJavaVersion` here, not `CURRENT` as the Gradle JVM and compiler JVM can be different (ex: passing `-Dcompiler.java=13`).